include/rmm/cuda_stream_view.hpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <cuda_runtime_api.h>
+
+#include <atomic>
+#include <cstddef>
+#include <cstdint>
+
+namespace rmm {
+
+/**
+ * @brief Strongly-typed non-owning wrapper for CUDA streams with default constructor.
+ *
+ * This wrapper is simply a "view": it does not own the lifetime of the stream it wraps.
+ */
+class cuda_stream_view {
+ public:
+  cuda_stream_view(cuda_stream_view const&) = default;
+  cuda_stream_view(cuda_stream_view&&)      = default;
+  cuda_stream_view& operator=(cuda_stream_view const&) = default;
+  cuda_stream_view& operator=(cuda_stream_view&&) = default;
+
+  // TODO disable construction from 0 after cuDF and others adopt cuda_stream_view
+  // cuda_stream_view(int)            = delete; //< Prevent cast from 0
+  // cuda_stream_view(std::nullptr_t) = delete; //< Prevent cast from nullptr
+
+  /**
+   * @brief Construct a default cudaStream_t.
+   */
+  constexpr cuda_stream_view() noexcept : stream_{0} {}
```

Comment:
Should this call `get_default_stream`?

Reply:
As discussed offline, `get_default_stream` (and `set_default_stream`) is no longer a thing.